base/namedtuple.jl

```diff
@@ -103,6 +103,8 @@ firstindex(t::NamedTuple) = 1
 lastindex(t::NamedTuple) = nfields(t)
 getindex(t::NamedTuple, i::Int) = getfield(t, i)
 getindex(t::NamedTuple, i::Symbol) = getfield(t, i)
+getindex(t::NamedTuple, v::AbstractVector{Symbol}) = [t[i] for i in v]
+getindex(t::NamedTuple, i) = values(t)[i]
```

Comment:
Can you explain what the semantics of this are? What are some examples where this helps you, and what do you expect as output?

Reply:
@ararslan suggested the latter as a fallback so that indexing methods for Tuple will be used.

Reply:
Indexing `NamedTuple`s only differs from that for `Tuple`s when indexing with `Symbol`s. Everything else should be identical, so we can get it all for free by using a fallback like this.